aptos-move/framework/aptos-framework/sources/transaction_validation.move

```diff
@@ -123,24 +126,28 @@ module aptos_framework::transaction_validation {
                 error::invalid_argument(PROLOGUE_ESEQUENCE_NUMBER_TOO_NEW)
             );
 
-            assert!(
-                txn_authentication_key == bcs::to_bytes(&transaction_sender),
-                error::invalid_argument(PROLOGUE_EINVALID_ACCOUNT_AUTH_KEY),
-            );
+            if (!features::simulation_enhancement_enabled() || !is_simulation || !is_auth_key_empty(&txn_authentication_key)) {
```

Comment:
Nit: We do this same set of checks on line 99, consider a helper function.

Reply:
Thank you! I refactored it with a helper function.